core/common/src/DateTimeUnit.kt

```diff
@@ -199,23 +209,14 @@ public sealed class DateTimeUnit {
         public val HOUR: TimeBased = MINUTE * 60
 
         /**
-         * A day.
+         * A calendar day.
          *
-         * We follow ISO-8601 in that we do not consider a day to be the same as 24 hours.
-         * The reason lies in time zone transitions, because of which some days can be 23 or 25 hours.
-         * For example, we say that exactly a whole day has passed between `2019-10-27T02:59` and `2019-10-28T02:59`
-         * in Berlin, despite the fact that the clocks were turned back one hour, so there are, in fact, 25 hours
-         * between the two date-times.
-         *
-         * It is for this reason that most operations with [DateBased] units require a [TimeZone], whereas those with
-         * [TimeBased] ones never do.
+         * Note that a calendar day is not the same as 24 hours.
```

Comment:
The reason why I placed the documentation here and not on `DayBased` is that I'd expect most people to never encounter `DayBased` and instead use `DAY` or its multiples. So, I think that the description of the calendar day should be at least referenced here.

Reply:
I've added a link from DAY to DayBased.